### PR TITLE
[sailjail] Allow specifying binary without path. Fixes JB#55089

### PR DESCRIPTION
**Pull request by Tomin1**
This relaxes checks for all apps and not just those run in compatibility
mode. If Exec line defines only application binary (+ arguments to app)
then it may be defined without path. In that case we require that the
invoked binary path points to a file in /usr/bin and basename matches to
Exec line's first word.

Rewrote the earlier implementation for legacy apps because it could have
mistakenly detected arguments for sailjail as the binary name. This
check is simpler and easier to read as it only requires to check the
first word of Exec line for binary name. Previously legacy apps could
have had sailjail wrapper although they weren't sandboxing aware.

### CODE REVIEW

File: daemon/sailjailclient.c [hunk 1]
```diff
@@ -1068,30 +1077,29 @@ sailjailclient_validate_argv(const char *exec, const gchar **app_argv, bool use_
...
-        *tpl_argv = g_strconcat(BINDIR "/", temp, NULL);
-        g_free(temp);
+        if( !*tpl_argv ) {
+            log_err("Exec line does not contain '%s'", *app_argv);
+            goto EXIT;
+        }
     }
 
-    if( !sailjailclient_match_argv(tpl_argv, app_argv) ) {
+    if( !sailjailclient_match_argv(tpl_argv+1, app_argv+1) ) {
```
Comment (Tomin1) [inline daemon/sailjailclient.c:1102]: I need to make some adjustments still. It this wants to check that the first word is not a field code so I have to adjust that to some direction.

Comment (spiiroin) [inline daemon/sailjailclient.c:1102]: nit: Should have space around '+' chars.

Also, while it is fairly obvious, one might add "arg zero already checked" type comment too.

**Review by spiiroin — APPROVED** [review]
LGTM.

As this touches code that is duplicated in mapplauncherd side: are they / do they need to be in sync regarding this compatibility bit?

File: daemon/sailjailclient.c (hunk 1 above)
Comment (Tomin1) [reply daemon/sailjailclient.c]: Removed that "Rule out template starting with a field code" test as redundant.

Comment (Tomin1) [reply daemon/sailjailclient.c]: Fixed. Also added that comment.

**Pull request merged**